uui/src/lib/TextField/TextField.svelte

```diff
@@ -48,9 +48,40 @@
     active = 'active';
   }
 
+  function getIconClass(): string {
+    let classes: string[] = [];
+    if ($$slots.leadingIcon) {
+      classes.push('mdc-text-field--with-leading-icon');
+    }
+    if ($$slots.trailingIcon) {
+      classes.push('mdc-text-field--with-trailing-icon');
+    }
+    return classes.join(' ');
+  }
+
+  function handleKeyEvt(e: KeyboardEvent) {
```

Comment:
Is there a more performant way to do this? like bubbling up from the component in the form rather that searching all forms etc...

Reply:
yeah, I was wondering if there was a way to tap into the "bubbling_phase" and see if it's on a form and then use that, but haven't figure that out yet. 

But there is this: 
`const form = (e.target as HTMLElement).closest('form');` which could be more performant. 

Could also use e.composedPath() to find the form in its event bubbling path. I'm not sure if closest() uses that, but seems like this is more manual work than needed since closest() is provided.